flask_static_digest/__init__.py

```diff
@@ -26,28 +26,29 @@ def init_app(self, app):
         app.config.setdefault("FLASK_STATIC_DIGEST_HOST_URL", None)
 
         self.host_url = app.config.get("FLASK_STATIC_DIGEST_HOST_URL")
-        self.static_url_path = app.static_url_path
 
-        self.manifest_path = os.path.join(app.static_folder,
-                                          "cache_manifest.json")
-        self.has_manifest = os.path.exists(self.manifest_path)
+        self.manifests = {}
 
-        self.manifest = {}
+        def ingestScaffold(endpoint, scaffold):
+            if not scaffold.has_static_folder:
+                return
 
-        if self.has_manifest:
-            self.manifest = self._load_manifest(app)
+            manifest_path = os.path.join(scaffold.static_folder,
+                                         "cache_manifest.json")
+            has_manifest = os.path.exists(manifest_path)
 
-        app.add_template_global(self.static_url_for)
+            if has_manifest:
+                manifest = self._load_manifest(scaffold, manifest_path)
+                self.manifests[endpoint] = manifest
 
-    def _load_manifest(self, app):
-        with app.open_resource(self.manifest_path, "r") as f:
-            manifest_dict = json.load(f)
+        ingestScaffold('static', app)
+        for endpoint, scaffold in app.blueprints.items():
+            ingestScaffold(endpoint + '.static', scaffold)
 
-        return manifest_dict
+        app.add_template_global(self.static_url_for)
 
-    def _prepend_host_url(self, host, filename):
-        return urljoin(self.host_url,
-                       "/".join([self.static_url_path, filename]))
+    def _prepend_host_url(self, flask_url):
+        return urljoin(self.host_url, flask_url)
```

Comment:
Was it intended to change this from its original behavior?

Reply:
Respectfully, the previous `"/".join([self.static_url_path, filename]` was a bad approach to creating urls which I ditched entirely. 
It reimplements the functionality of flask's url_for and wasn't going support routing to blueprints moving forward. I've removed it in favor of always relying on flask's `url_for`. very much intentional.

More thoughts on that in the next conversation item.

Reply:
I vaguely remember there being a problem when the `host_url` was more than a domain such as `cdn.example.com`. For example if you had `cdn.example.com/custom/path` I think `urljoin` didn't work out of the box. Or maybe it was a problem when the `static_url_path` had nested URL paths. It was like 4 years ago.

The general problem was making `urljoin` work with multiple parts. I'm pretty sure I got that implementation from https://stackoverflow.com/questions/36189261/python-join-multiple-components-to-build-a-url.

Reply:
Well the good news is you were already joining paths using urljoin so no regression here.

If flask's `url_for` can't produce the correct url, someone is undoubtedly misusing flask and we shouldn't be expected to support that. I don't care if it breaks for people if it's in line with Flask's expected behavior.

Reply:
I think originally I had the extra `join` in there so the extra parts were handling by the join and then `urljoin` only has to deal with 2 parts since the `static_url_path` and filename parts were handled by the `join`.

For example, here's what happens when you don't do the extra `join` and you have more than 2 parts:

```py
>>> urljoin("https://cdn.example.com", "hello", "app.css")
'https://cdn.example.com/hello'
```

`urljoin` by itself will not support 3 arguments in the way we want.

But my original approach does:

```py
>>> urljoin("https://cdn.example.com", "/".join(["hello", "app.css"]))
'https://cdn.example.com/hello/app.css'
```

It also supports the extra slash if it happens to exist:

```py
>>> urljoin("https://cdn.example.com", "/".join(["/hello", "app.css"]))
'https://cdn.example.com/hello/app.css'
```

But I guess with your new implementation, `flask_url` already handles joining the `static_url_path` and the asset's file name so `urljoin` is only ever dealing with 2 parts so it works? I dropped in a couple of prints and see `flask_url` is equal to `/static/css/app-d41d8cd98f00b204e9800998ecf8427e.css`.

I think we're good with leaving the new version.